tests/unit/asyn/wallet/test_main.py

```diff
@@ -0,0 +1,320 @@
+        "seed": "ssL9dv2W5RK8L3tuzQxYY6EaZhSxW",
+        "ed25519": {
+            "public_key": "ED16FD02F7A7E52B6ACB35F8A4D7013DC94755951629DB6611483590AC0E9FC6D5",  # noqa:E501
+            "private_key": "ED713A8C3171A3D8F69FE9526D9243834B3A30BF893B1EBC1824B1D96F18B44DCF",  # noqa:E501
+            "classic_address": "rPxLJ2GumwfLk3z9njmmpo8nAX7sfZYptV",
+        },
+        "secp256k1": {
+            "public_key": "030E58CDD076E798C84755590AAF6237CA8FAE821070A59F648B517A30DC6F589D",  # noqa:E501
+            "private_key": "00141BA006D3363D2FB2785E8DF4E44D3A49908780CB4FB51F6D217C08C021429F",  # noqa:E501
+            "classic_address": "rhvh5SrgBL5V8oeV9EpDuVszeJSSCEkbPc",
+        },
+    },
+    "entropy": {
+        "entropy": "00000000000000000000000000000000",
+        "ed25519": {
+            "seed": "sEdSJHS4oiAdz7w2X2ni1gFiqtbJHqE",
+            "public_key": "ED1A7C082846CFF58FF9A892BA4BA2593151CCF1DBA59F37714CC9ED39824AF85F",  # noqa:E501
+            "private_key": "ED0B6CBAC838DFE7F47EA1BD0DF00EC282FDF45510C92161072CCFB84035390C4D",  # noqa:E501
+            "classic_address": "r9zRhGr7b6xPekLvT6wP4qNdWMryaumZS7",
+        },
+        "secp256k1": {
+            "seed": "sp6JS7f14BuwFY8Mw6bTtLKWauoUs",
+            "public_key": "0390A196799EE412284A5D80BF78C3E84CBB80E1437A0AECD9ADF94D7FEAAFA284",  # noqa:E501
+            "private_key": "002512BBDFDBB77510883B7DCCBEF270B86DEAC8B64AC762873D75A1BEE6298665",  # noqa:E501
+            "classic_address": "rGCkuB7PBr5tNy68tPEABEtcdno4hE6Y7f",
+        },
+    },
+    "prefix": {
+        "classic_address": "r",
+        "ed25519_key": "ED",
+        "secp256k1_private_key": "00",
+    },
+    "xaddress": {
+        "tag": 1337,
+        "mainnet_xaddress": "X7gJ5YK8abHf2eTPWPFHAAot8Knck11QGqmQ7a6a3Z8PJvk",
+        "testnet_xaddress": "T7bq3e7kxYq9pwDz8UZhqAZoEkcRGTXSNr5immvcj3DYRaV",
+        "wallet": Wallet(
+            "030E58CDD076E798C84755590AAF6237CA8FAE821070A59F648B517A30DC6F589D",
+            "00141BA006D3363D2FB2785E8DF4E44D3A49908780CB4FB51F6D217C08C021429F",
+        ),
+    },
+}
+
+
+def _test_wallet_fields(
+    self, wallet, method, algorithm, *, is_entropy_with_regular_key_pair=False
+):
+    method_constants = constants[method]
+    algorithm_constants = method_constants[algorithm]
+
+    self.assertEqual(wallet.public_key, algorithm_constants["public_key"])
+    self.assertEqual(wallet.private_key, algorithm_constants["private_key"])
+
+    if method == "entropy":
+        self.assertEqual(wallet.seed, algorithm_constants["seed"])
+    else:
+        self.assertEqual(wallet.seed, method_constants["seed"])
+
+    if method == "regular_key_pair" or is_entropy_with_regular_key_pair:
+        self.assertEqual(
+            wallet.classic_address, constants["regular_key_pair"]["master_address"]
+        )
+    else:
+        self.assertEqual(wallet.classic_address, algorithm_constants["classic_address"])
+
+
+def _test_wallet_types(self, wallet, algorithm):
```

Comment:
This seems very similar to the other test format name. Can we be a bit more specific?

Ex.
```suggestion
def _test_wallet_encoding_algorithm(self, wallet, algorithm):
```

Reply:
`_test_wallet_fields` is for checking the actual values of the wallet fields

`_test_wallet_types` if for checking the actual type of the wallet fields

I don't think `_test_wallet_encoding_algorithm` makes sense here. Would it be better to change `_test_wallet_fields` to `_test_wallet_values`?

Reply:
Yep that makes sense to me